### ISSUE TITLE
Allow to store an "audit user" for changes via restAPI

### ISSUE THREAD

**Issue by JohannesKreuzer** [netbox-community/netbox#18055]
### NetBox version

v4.1.6

### Feature type

Data model extension

### Triage priority

N/A

### Proposed functionality

allow to pass an username (actually an arbitrary string) via restAPI to store the user of the upstream application which triggered the rest api request for auditing purposes, it should default to the username of the netbox user triggering the request, if it is not passed in the request. Might be done via an additional HTTP-Header to be sent from restAPI client to netbox server.

### Use case

Netbox is used as a Datasource for an upstream application. To be able to see who really triggered the changes in netbox, it is needed to be able to pass the username from the upstream application to netbox.

### Database changes

add an additonal audit_username field of type string to ChangeLog model.

### External dependencies

support in the netbox restAPI client libraries

**Comment by JohannesKreuzer** [comment]
@rboucher-me this is not about having different applications reusing the API key, but about to be able to audit which user of the application triggered the changes in NetBox.

**Comment by rboucher-me** [comment]
Understood. You make a good point and the "audit trail" use case probably requires further consideration.

**Comment by jeremystretch** [comment]
> to be able to audit which user of the application triggered the changes in NetBox.

The application would need to be configured to act as that user, with an API token tied to the user's account in NetBox.

**Comment by JohannesKreuzer** [comment]
@jeremystretch 
In the upstream application, you don't want every user (which is unaware of NetBox, and would have no direct access to NetBox) to have to go to NetBox to create an API token and feed to the upstream Application and store there.

e.g. think about a Managed hosting system, managing VMs on the hypervisor and also storing the information in NetBox. Users are only allowed to access the application itself, but have no access to the hypervisor and NetBox. Only the application has access with restricted service credentials, only granting the least necessary privileges to the application. The changes in NetBox should then be traceable to the application user which triggered the change (creating a tuple to identify the original user (NetBox User + audit user).

**Comment by jeremystretch** [comment]
> In the upstream application, you don't want every user (which is unaware of NetBox, and would have no direct access to NetBox) to have to go to NetBox to create an API token and feed to the upstream Application and store there.

It seems you're conflating two authentication systems. If the user of a remote application has no equivalent account in NetBox, then it makes no sense for NetBox to associate that user with a change made in NetBox: That information would need to be recorded within the remote application. NetBox knows only about the users which have accounts created within it.

**Comment by JohannesKreuzer** [comment]
@jeremystretch We don't want to associate an additional NetBox user, but from DB perspective a string field in the ChangeLog entry, to store this info.

**Comment by jeremystretch** [comment]
Right; a string that represents a user completely foreign to NetBox. It's arbitrary metadata, extraneous to the intended purpose of change logging in NetBox.

**Comment by jnovinger** [comment]
While we can appreciate this and related use cases, we don't think this is the proper way to attack this. This points to a larger feature that would require more exploration and design work.
